### DISCUSSION TITLE
Support multiple network interfaces for NAT1To1IPs

### DISCUSSION THREAD

**Discussion by craymond12** [pion/webrtc#2802] [Ideas]
Currently, if a device hardware have multiple network interfaces, SetNAT1To1IPs can only be used for one of them.
But it is not always known in advance which network interface can allow host to host connection
This is mostly used with SetICETCPMux

Is there any technical reason why the external_ip_mapper.go prevent the mapping of multiple external IP to the same local IP?

**Comment by craymond12** [comment]
If you think that it should possible, I can look at it more in depth and do a PR to support it

**Comment by Sean-Der** [comment]
Hey @craymond12 

Sorry I didn't respond to this sooner. Continuing our conversation from Slack, I think this is possible today!

The external IP mapper does splitting [here](https://github.com/pion/ice/blob/master/external_ip_mapper.go#L95-L98) it does `external_ip/local_ip`

If you wouldn't mind making a PR to improve the documentation I would really appricate it

**Reply by craymond12** [reply]
no problem
I should be testing this soon and see how it goes

Meanwhile I will try to think at how to improve the documentation of that function to make it clearer for others
